ios/sdk/src/CallKit/JMCallKitManager.swift

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright @ 2017-present Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CallKit
+import Foundation
+
+
+/// JitsiMeet CallKit proxy
+@objc public final class JMCallKitManager: NSObject {
+    
+    override private init() {}
+    
+    // MARK: - CallKit proxy
+    
+    internal static let cxProvider: CXProvider = {
+        let config = CXProviderConfiguration(localizedName: "")
+        let provider = CXProvider(configuration: config)
+        return provider
+    }()
+    
+    internal static let cxCallController: CXCallController = {
+        return CXCallController()
+    }()
+    
+    internal static let callKitNotifier: JMCallKitNotifier = {
+        return JMCallKitNotifier()
+    }()
+    
+    internal static var cxProviderConfiguration: CXProviderConfiguration? {
+        didSet {
+            guard let providerConfiguration = cxProviderConfiguration else { return }
+            cxProvider.configuration = providerConfiguration
+            cxProvider.setDelegate(callKitNotifier, queue: nil)
+        }
+    }
+    
+    @objc public static func hasProviderBeenConfigurated() -> Bool {
+        return cxProviderConfiguration != nil
+    }
+        
+    @objc public static func configureCallKitProvider(localizedName: String,
+                                                      ringtoneSound: String?,
+                                                      iconTemplateImageData: Data?,
+                                                      includesCallsInRecents: Bool) {
+        let configuration = CXProviderConfiguration(localizedName: localizedName)
+        configuration.ringtoneSound = ringtoneSound
+        configuration.iconTemplateImageData = iconTemplateImageData
+        //configuration.includesCallsInRecents = includesCallsInRecents
```

Comment:
yes, only enabled for iOS 11. Maybe someone wants to prefer using it as false because they haven't implemented handling intents? I'm fine just removing the option to configure this setting